sdks/java/core/src/main/java/org/apache/beam/sdk/util/BigQueryTableInserter.java

```diff
@@ -82,8 +80,8 @@
   private final TableReference defaultRef;
   private final long maxRowsPerBatch;
 
-  private static final ExecutorService executor = MoreExecutors.getExitingExecutorService(
-      (ThreadPoolExecutor) Executors.newFixedThreadPool(100), 10, TimeUnit.SECONDS);
+  private static final ExecutorService executor =
+          new GcsOptions.ExecutorServiceFactory().create();
```

Comment:
is it possible to initialize this on first use rather than when the class is loaded into memory?